third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -152,6 +152,12 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::GetDataShards(
       shards.push_back(std::make_shared<PjRtData>(
           shard->device(), shard->shape(), shard->buffer));
     }
+  } else if (PjRtReplicatedData* replicated_data =
```

Comment:
This is piggybacking on `GetDataShards`, so it might be better to move this to a new ComputationClient function. I'm happy to make that change

Reply:
Looking at this dynamic cast & check, I am thinking maybe we should use the PjRtShardedData to hold the replicated data to track. PjRtShardedData can have a flag `replicated` or we can supply OpSharding::Replicated to it (it already holds the sharding spec) for lookup. This way, we don't need to extra auxiliary APIs, like `WrapReplicatedData`, and just re-use the existing similar APIs.